frontend/src/shared/helpers.tsx

```diff
@@ -512,6 +510,33 @@ export function prepareCohortDataForAgGrid(
     });
   });
 
+  if (sortModel) {
+    const sortField = Object.keys(
+      sortModel[0]
+    )[0] as keyof typeof newCohorts[number];
+    const sortOrder = sortModel[0][sortField];
+    newCohorts.sort((objA, objB) => {
+      let a = objA[sortField],
+        b = objB[sortField];
+
+      if (a === null || a === undefined) return 1;
+      if (b === null || b === undefined) return -1;
+
+      if (Array.isArray(a)) a = a.join(", ");
+      if (Array.isArray(b)) b = b.join(", ");
+
+      if (typeof a === "number" && typeof b === "number") {
+        return sortOrder === "ASC" ? a - b : b - a;
+      } else if (typeof a === "string" && typeof b === "string") {
+        return sortOrder === "ASC"
+          ? a.localeCompare(b, "en", { sensitivity: "base" })
+          : b.localeCompare(a, "en", { sensitivity: "base" });
+      } else {
+        return 0;
+      }
+    });
+  }
```

Comment:
Handle sorting for Cohorts view where we're filtering and now sorting from the frontend. Initially, I planned on moving all of this logic of the Cohorts view to the GraphQL server, but I'll save that work for [this card](https://app.zenhub.com/workspaces/metadb-scrum-5fcfa28a4ccc1600165347b6/issues/gh/mskcc/smile-server/1229).